src/CHIPDriver.hh

```diff
@@ -118,4 +120,183 @@ extern std::atomic_ulong CHIPNumRegisteredFatBinaries;
+    logDebugSettings();
+  }
+
+  int getPlatformIdx() const { return PlatformIdx_; }
+  DeviceType getDevice() const { return Device_; }
+  int getDeviceIdx() const { return DeviceIdx_; }
+  BackendType getBackend() const { return Backend_; }
+  bool getDumpSpirv() const { return DumpSpirv_; }
+  const std::string &getJitFlags() const { return JitFlags_; }
+  bool getL0ImmCmdLists() const { return L0ImmCmdLists_; }
+  int getL0CollectEventsTimeout() const { return L0CollectEventsTimeout_; }
+
+private:
+  void parseEnvironmentVariables() {
+    // Parse all the environment variables and set the class members
+    if (!readEnvVar("CHIP_PLATFORM").empty()) {
+      PlatformIdx_ = parseInt("CHIP_PLATFORM");
+      Device_ = parseDeviceType("CHIP_DEVICE_TYPE");
+    }
+    if (!readEnvVar("CHIP_DEVICE").empty()) {
+      DeviceIdx_ = parseInt("CHIP_DEVICE");
+      Backend_ = parseBackendType("CHIP_BE");
+    }
+    if (!readEnvVar("CHIP_DUMP_SPIRV").empty()) {
+      DumpSpirv_ = parseBoolean("CHIP_DUMP_SPIRV");
+      JitFlags_ = parseJitFlags("CHIP_JIT_FLAGS_OVERRIDE");
+    }
+    if (!readEnvVar("CHIP_L0_IMM_CMD_LISTS").empty())
+      L0ImmCmdLists_ = parseBoolean("CHIP_L0_IMM_CMD_LISTS");
+    if (!readEnvVar("CHIP_L0_COLLECT_EVENTS_TIMEOUT").empty())
+      L0CollectEventsTimeout_ = parseInt("CHIP_L0_COLLECT_EVENTS_TIMEOUT");
+  }
+
+  std::string_view parseJitFlags(const std::string &StrIn) {
+    if (readEnvVar(StrIn).empty())
+      return CHIP_DEFAULT_JIT_FLAGS;
+
+    return JitFlags_;
+  }
+
+  int parseInt(const std::string &StrIn) {
+    const auto &Str = readEnvVar(StrIn);
+    if (!isConvertibleToInt(Str))
+      CHIPERR_LOG_AND_THROW("Invalid integer value: " + Str,
+                            hipErrorInitializationError);
+    return std::stoi(str);
+  }
+
+  bool parseBoolean(const std::string &StrIn) {
+    auto str = readEnvVar(StrIn);
+    if (str == "1" || str == "on")
+      return true;
+    if (str == "0" || str == "off")
+      return false;
+    CHIPERR_LOG_AND_THROW("Invalid boolean value: " + str + "while parsing " +
+                              StrIn,
+                          hipErrorInitializationError);
+    return false; // This return is never reached
+  }
+
+  DeviceType parseDeviceType(const std::string &StrIn) {
+    auto str = readEnvVar(StrIn);
+    if (str == "gpu")
+      return DeviceType(DeviceType::GPU);
+    if (str == "cpu")
+      return DeviceType(DeviceType::CPU);
+    if (str == "accel")
+      return DeviceType(DeviceType::Accelerator);
+    if (str == "fpga")
+      return DeviceType(DeviceType::FPGA);
+    if (str == "")
+      return DeviceType(DeviceType::Default);
+
+    CHIPERR_LOG_AND_THROW("Invalid device type value: " + str +
+                              " while parsing " + StrIn,
+                          hipErrorInitializationError);
+    return DeviceType(DeviceType::GPU); // This return is never reached
+  }
+
+  BackendType parseBackendType(const std::string &StrIn) {
```

Comment:
Same as for the parseDeviceType(). This function looks like it belongs to the BackendType class.